### PR TITLE
Improvements and Android Q support

### PR DESCRIPTION
**Pull request by enisinanaj**
# Summary

There is an issue in the library regarding file handling on Android 10+.
The issue comes with the introduction of Scoped Storage in Android. This update makes it possible to save the file in the gallery even on an Android 10+ system.

### What's required for testing (prerequisites)?

In AndroidManifest.xml be sure to add the property `android:requestLegacyExternalStorage="true"` to true for the `application` tag.

Android 11 and Android 10 with the above property set in the application tag will work fine and will be able to access the file's content through a direct file path and the File java API.

### What are the steps to reproduce (after prerequisites)?

Save a media file or document to the gallery in an Android device using Android 10+

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Android |   ✅     |

## Checklist

- [ x] I have tested this on a device using browser stack and on a simulator  (something might have slipped. please verify if possible)

### CODE REVIEW

**Comment by enisinanaj** [comment]
#244

File: android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            }
+        }
+    }
+
+    private void manageAndroidQ(File source) throws IOException {
+        FileChannel input = new FileInputStream(source).getChannel();
+        String fullSourceName = source.getName();
+
+        ContentResolver resolver = mContext.getContentResolver();
+        Uri videoCollection = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;;
```
Comment (rishabh876) [inline android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java:137]: This seems to be a video specific solution.

File: android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+        ContentValues contentDetails = new ContentValues();
+        contentDetails.put(MediaStore.Video.Media.DISPLAY_NAME, fullSourceName);
+        contentDetails.put(MediaStore.Video.Media.IS_PENDING, 1);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            String relativePath = Environment.DIRECTORY_MOVIES + File.separator + mContext.getPackageName();
```
Comment (rishabh876) [inline android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java:143]: Shouldn't `mContext.getPackageName()` be album name provided.

Comment (enisinanaj) [reply android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java:143]: for what I read it has to be the package name of the application and of course the album name should be added to it. that's something missing.

File: android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java (hunk 1 above)
Comment (enisinanaj) [reply android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java:137]: yeah you're totally right

Comment (enisinanaj) [reply android/src/main/java/com/reactnativecommunity/cameraroll/operation/SaveToCameraRoll.java:137]: thanks for going through it mate

**Comment by MadCoderme** [comment]
Please solve the issues and merge it as soon as possible, we are really waiting for this!

**Comment by idrissakhi** [comment]
Resolved in v5.0.0

**Comment by idrissakhi** [comment]
Resolved in v5.0.0

**Pull request closed without merging**